jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/JGivenPlugin.java

```diff
@@ -79,20 +78,18 @@ private void addDefaultReports(final Project project) {
 
     private void configureDefaultReportTask(final Test test, JGivenReportTask reportTask,
                                             final ReportingExtension reportingExtension) {
-        reportTask.mustRunAfter(test);
-
-        ConventionMapping mapping = ((IConventionAware) reportTask).getConventionMapping();
-        Callable<File> getResultsDirectory = () -> test.getExtensions()
-            .getByType(JGivenTaskExtension.class)
-            .getResultsDir();
-        mapping.map("results", getResultsDirectory);
-
-        Objects.requireNonNull(mapping.getConventionValue(reportTask.getReports(), "reports", false))
-            .all(report -> {
-                ConventionMapping reportMapping = ((IConventionAware) report).getConventionMapping();
-                String relativeFilePath = "jgiven" + "/" + test.getName() + "/" + report.getName();
-                Callable<File> getDestination = () -> reportingExtension.file(relativeFilePath);
-                reportMapping.map("destination", getDestination);
-            });
+
+        Provider<Directory> getResultsDirectory = test.getExtensions()
+                .getByType(JGivenTaskExtension.class)
+                .getResultsDir();
+
+        reportTask.getResults().set(getResultsDirectory);
```

Comment:
```suggestion
        Provider<Directory> resultsDirectory= test.getExtensions()
                .getByType(JGivenTaskExtension.class)
                .getResultsDir();

        reportTask.getResults().convention(resultsDirectory);
```

Using `convention` makes this the "fall-back" value. Then you can be sure that if a user re-configures this it will override the default. Changing the value here also allows users to "break" the task dependency to the test task if they are not happy with that change.

Reply:
Funnily enought, I had this as convention for a while, but convinced myself that "set" was correct ^^